### PR TITLE
NAS Benchmark integration (stage 1)

### PR DESCRIPTION
**Pull request by ultmaster**
This PR will integrate NAS-Bench-101 and 201 benchmark data into Retiarii search space, to enable query "real" architectures on-the-fly.

### CODE REVIEW

File: nni/retiarii/execution/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+class BenchmarkExecutionEngine(BaseExecutionEngine):
+    """
+    Execution engine that does not actually run any trial, but query the database for results.
+
+    The database query is done on the trial end to make sure intermediate metrics are available.
+    It also supports an accelerated mode that returns metric immediately without even running into NNI manager.
+    """
+
+    def __init__(self, benchmark: Union[str, Callable[[BenchmarkGraphData], Tuple[float, List[float]]]], acceleration: bool = False):
```
Comment (QuanluZhang) [inline nni/retiarii/execution/benchmark.py:51]: `acceleration` is not used?

File: nni/retiarii/execution/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        super().__init__()
+        assert benchmark in BenchmarkGraphData.SUPPORTED_BENCHMARK_LIST, \
+            f'{benchmark} is not one of the supported benchmarks: {BenchmarkGraphData.SUPPORTED_BENCHMARK_LIST}'
+        self.benchmark = benchmark
+        self.acceleration = acceleration
+
+    def pack_model_data(self, model: Model) -> Any:
```
Comment (QuanluZhang) [inline nni/retiarii/execution/benchmark.py:58]: where is `pack_model_data` invoked? better to add docstring for it

File: nni/retiarii/execution/benchmark.py (hunk 1 above)
Comment (ultmaster) [reply nni/retiarii/execution/benchmark.py:51]: This is left to "stage 2".

File: nni/retiarii/execution/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+def _flatten_architecture(mutation: Dict[str, Any], benchmark: Optional[str] = None):
+    # STRONG ASSUMPTION HERE!
+    # This assumes that the benchmarked search space is a one-level search space.
+    # This means that it is either ONE cell or ONE network.
+    # Two cell search space like NDS is not supported yet for now.
+    # Some benchmark even needs special handling to pop out invalid keys. I don't think this is a good design.
+
+    # support double underscore to be compatible with naming convention in base engine
+    ret = {k.split('/')[-1].split('__')[-1]: v for k, v in mutation.items()}
+    if benchmark == 'nasbench101':
```
Comment (QuanluZhang) [inline nni/retiarii/execution/benchmark.py:134]: why "nasbench101" is a special case?

Comment (QuanluZhang) [reply nni/retiarii/execution/benchmark.py:134]: let's improve arch format in the following pr

**Review by QuanluZhang — APPROVED** [review]

**Review by skeletondyh — APPROVED** [review]

**Review by acured — APPROVED** [review]

**Pull request merged**